src/components/head-metadata/index.tsx

```diff
@@ -0,0 +1,52 @@
+import HashiHead from '@hashicorp/react-head'
+import { useCurrentProduct } from 'contexts'
+import { HeadMetadataProps } from './types'
+
+/**
+ * Builds up the the necessary meta tags for the site. Rendered in `_app`, where it receives `pageProps.metadata` as props
+ *
+ * We build up a page title that looks like {props.title} | {currentProduct} | {root title}
+ */
+export default function HeadMetadata(props: HeadMetadataProps) {
+  const { name: productName, slug: productSlug } = useCurrentProduct() ?? {}
+
+  const titleParts = [__config.dev_dot.meta.title]
+  const description = props.description ?? __config.dev_dot.meta.description
+
+  // We're using .unshift() to add the following elements to the beginning of the array
+  if (productName) {
+    titleParts.unshift(productName)
+  }
+
+  if (props.title) {
+    titleParts.unshift(props.title)
+  }
+
+  const title = titleParts.join(' | ')
+
+  const finalDescription = description.replace(
+    '{product}',
+    productName ?? 'HashiCorp'
+  )
+
+  const ogImageUrl = `/og-image/${productSlug ?? 'base'}.jpg`
```

Comment:
I think this is pretty low level to be including this logic, but I am fine with it as it will be easy to work away from in the event we need something more flexible.

I think the two scenarios I am thinking about:
- Adding share cards for other products, we'll probably have to dig in this code to see how they are referenced as it may not be super obvious as we roll out the other products
- In the event we decide to roll something with a different set of logic than just per-product / base share cards, this will have to be changed.

Just thinking out loud here, for the most part.

Reply:
Good call. I think this is a reasonable initial implementation, we can thread through an `image` property much like we do `title` and `description`, from pages when we get to the point where we need more granular customization. I'll add a note to the README section explaining where we pull images from.

Reply:
Rad, that more than satisfies me 👌 we can wait for the need to arise to refactor if need be if we need more flexability.